File: backend/app/schemas/selection.py
from datetime import date

from pydantic import BaseModel

from app.schemas.tools import (
    InventoryPressureOutput,
    InventoryStatusOutput,
    RestockOutput,
    SalesMetricsOutput,
    SalesTrendOutput,
)


class SelectionAnalysis(BaseModel):
    site: str
    platform: str
    start_date: date
    end_date: date
    product_names: list[str]
    sales: SalesMetricsOutput
    trend: SalesTrendOutput | None
    inventory: InventoryStatusOutput
    pressure: InventoryPressureOutput
    restock: RestockOutput
    average_daily_sales: float
    risks: list[str]
    evidence: list[str]
    human_confirmation_notice: str = "补货建议必须由人工确认，MVP 不执行真实采购。"

File: frontend/src/types/selection.ts
export type SelectionAnalysis = {
  site: string;
  platform: string;
  start_date: string;
  end_date: string;
  product_names: string[];
  sales: {
    units_sold: number;
    revenue: number;
    gross_margin_rate: number;
    refund_rate: number;
  };
  trend: {
    direction: "growing" | "stable" | "declining";
    overall_growth_rate: number;
    latest_mom_rate: number;
  } | null;
  inventory: {
    items: Array<{
      sku: string;
      product_name: string;
      snapshot_date: string;
      available: number;
      inbound: number;
    }>;
  };
  pressure: {
    coverage_days: number | null;
    pressure: string;
    rationale: string;
  };
  restock: {
    should_restock: boolean;
    recommended_quantity: number;
    rationale: string;
    requires_human_confirmation: boolean;
  };
  risks: string[];
  evidence: string[];
  human_confirmation_notice: string;
};
